Detectors/TRD/base/include/TRDBase/Digit.h

```diff
@@ -13,17 +13,87 @@
 
 #include <cstdint>
 #include <vector>
+#include <array>
+#include <map>
 
 namespace o2
 {
 namespace trd
 {
 
-class DigitIndex;
+class Digit;
 
-typedef std::uint16_t Digit;
-typedef std::vector<Digit> DigitContainer;
-typedef std::vector<DigitIndex> DigitIndexContainer;
+constexpr int kTB = 30;
+constexpr int KEY_MIN = 0;
+constexpr int KEY_MAX = 2211727;
+
+typedef std::uint16_t ADC_t;                              // the ADC value type
+typedef std::array<ADC_t, kTB> ArrayADC_t;                // the array ADC
+typedef std::vector<Digit> DigitContainer_t;              // the digit container type
```

Comment:
not sure having typedefs will help readability much. (std::vector<Digit> is as long as DigitContainer_t and tells me more when I see it). Personal taste.

Reply:
Over the last few days, we were changing this a lot and therefore replacing containers and classes everywhere, this way it was easier and makes it more flexible to implement changes in the future. In particular `ADC_t` should remain a `typedef` as future testing might require it to be a `float`. For the containers, it is personal taste too to keep the types as `*Container_t`.